sweet/harnesses/cockroachdb.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2024 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package harnesses
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"golang.org/x/benchmarks/sweet/common"
+	"golang.org/x/benchmarks/sweet/common/log"
+)
+
+type CockroachDB struct{}
+
+func (h CockroachDB) CheckPrerequisites() error {
+	return nil
+}
+
+func (h CockroachDB) Get(gcfg *common.GetConfig) error {
+	// Build against the latest stable release.
+	return gitShallowClone(
+		gcfg.SrcDir,
+		"https://github.com/cockroachdb/cockroach",
+		"release-24.1",
```

Comment:
Maybe `master` is better, otherwise someone has to periodically update the next release branch?

Reply:
True, but leaving it as master would mean that it might occasionally break outside of this benchmark's control. I see that the `etcd` picks a specific release and leaves a comment to try and keep it close to HEAD. 

I changed it to `v24.1.0-rc.1` instead for now, and maybe by the time this benchmark is ready we can switch to an actual dot release? But maybe final say is up to the Go team anyway, so I'm okay with whatever for now.